src/cmds/qstat.c

```diff
@@ -294,6 +309,47 @@ states(char *string, char *q, char *r, char *h, char *w, char *t, char *e, int l
 	}
 }
 
+/**
+ * @brief
+ *	Prepend escape character with special characters such as line feed and form feed
+ *
+ * @param[in]   val - string which needs to be formatted
+ *
+ * @return string
+ * @retval NULL     - if memory allocation failed.
+ * @retval string   - formatted string.
```

Comment:
I'd rather prefer leaving the caller with less responsibility. Just freeing up the returned string in this case.  I can see many utility functions where a string is returned and passing this responsibility of freeing up the string buffer to the caller. I'll follow the same convention.